hudi-common/src/test/java/org/apache/hudi/common/table/read/TestHoodieFileGroupReaderBase.java

```diff
@@ -668,7 +668,7 @@ private void validateOutputFromFileGroupReaderWithNativeRecords(StorageConfigura
         .map(r -> HoodieAvroUtils.removeFields(r, metaCols))
         .collect(Collectors.toSet());
     Set<GenericRecord> expectedRecordSet = expectedRecords.stream()
-        .map(r -> (GenericRecord) r.getRight())
+        .map(r -> resetByteBufferPosition((GenericRecord) r.getRight()))
```

Comment:
Is this needed for production code?

Reply:
This happens when we write out the records. There are some steps after the write on the write handle but we don't do anything special today for this so I am not sure if it is not required or a bug.